atlasdb-client/src/main/java/com/palantir/atlasdb/transaction/impl/TimestampCache.java

```diff
@@ -0,0 +1,41 @@
+/**
+ * Copyright 2016 Palantir Technologies
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.palantir.atlasdb.transaction.impl;
+
+import com.google.common.cache.Cache;
+import com.google.common.cache.CacheBuilder;
+
+// This class just here for readability and not directly leaking / tying us down to a Guava class in our API
+public class TimestampCache {
+    final Cache<Long, Long> timestampCache;
+
+    public TimestampCache() {
+        timestampCache = CacheBuilder.newBuilder()
+                .maximumSize(1_000_000) // up to ~32MB with java Long object bloat
```

Comment:
@tjwilson90 curious how you arrived at that, though I think on 64-bit with compressed OOPs we'll actually be around 72 bytes per entry, so 72MB as Cache with max size shows `Cache_MaxSize :: Bytes =  72.24, Objects =  2.00 Refs =  8.06 Primitives = {int=1.0, long=1.0}` in local version of https://github.com/DimitrisAndreou/memory-measurer/blob/master/ElementCostInDataStructures.txt